lib/apiservers/engine/backends/container.go

```diff
@@ -289,7 +298,7 @@ func (c *Container) ContainerExecInspect(eid string) (*backend.ExecInspect, erro
 	// Look up the container name in the metadata cache to get long ID
 	vc := cache.ContainerCache().GetContainerFromExec(eid)
 	if vc == nil {
-		return nil, NotFoundError(eid)
+		return nil, InternalServerError(fmt.Sprintf("No container was found with exec id: %s", eid))
```

Comment:
Why not use the `NotFoundError`? This does seem like a 404 - presumably an incorrect `eid` was supplied.

Reply:
When we look into the `NotFoundError` function it is actually prefacing it with "Container Not Found". What I will do is create my own custom `TaskNotFoundError` function instead. and place it in the same place as the `NotFoundError` function.